ext/nnstreamer/tensor_filter/tensor_filter_tvm.cc

```diff
@@ -0,0 +1,413 @@
+  bool invalid_option = false;
+
+  if (custom_prop == nullptr) {
+    /* no custom properties */
+    return true;
+  }
+
+  options = g_strsplit (custom_prop, ",", -1);
+  
+  for (guint op = 0; op < g_strv_length (options); ++op) {
+    gchar **option = g_strsplit (options[op], ":", -1);
+
+    if (g_strv_length (option) > 1) {
+      g_strstrip (option[0]);
+      g_strstrip (option[1]);
+
+      if (g_ascii_strcasecmp (option[0], "device") == 0) {
+        if (g_ascii_strcasecmp (option[1], "CPU") == 0) {
+          device = DLDevice{kDLCPU, 0};
+        } else if (g_ascii_strcasecmp (option[1], "GPU") == 0) {
+          device = DLDevice{kDLGPU, 0};
+        } else {
+          nns_loge ("Unknown device (%s).", option[1]);
+          invalid_option = true;
+          
+        }
+      } else {
+        nns_logw ("Unknown option (%s).", options[op]);
+      }
+    }
+    if (invalid_option)
+      break;
+    g_strfreev (option);
+  }
+  g_strfreev (options);
+  return !invalid_option;
+}
+
+/**
+ * @brief Configure tvm instance
+ */
+void tvm_subplugin::configure_instance (const GstTensorFilterProperties *prop)
+{
+  if (!parse_custom_prop (prop->custom_properties)) {
+    nns_loge ("Failed to parse custom property.");
+    throw std::invalid_argument ("Failed to parse custom property.");
+  }
+  
+  if (!empty_model) {
+    if (!prop->model_files[0] || prop->model_files[0][0] == '\0') {
+      std::cerr << "Model path is not given." << std::endl;
+      throw std::invalid_argument ("Model path is not given.");
+    }
+    cleanup ();
+  }
+
+  /* read model */
+  model_path = g_strdup (prop->model_files[0]);
+  mod_factory = tvm::runtime::Module::LoadFromFile (model_path, "so");
+  gmod = mod_factory.GetFunction ("default") (device);
+
+  if (!set_tensor_info (inputInfo, prop->input_meta, (int) gmod.GetFunction ("get_num_inputs") ()) || 
+      !set_tensor_info (outputInfo, prop->output_meta, (int) gmod.GetFunction ("get_num_outputs") ())) {
+    nns_loge ("Failed to set tensor info.");
+    throw std::invalid_argument ("Failed to set tensor info.");
+  }
+  if (!configure_meta (inputInfo, input_info_list) ||
+      !configure_meta (outputInfo, output_info_list)) {
+    nns_loge ("Failed to configure tensor meta.");
+    throw std::invalid_argument ("Failed to configure tensor meta.");
+  }
+  empty_model = false;
+}
+
+/**
+ * @brief Internal method to set tensor information for model input & output
+ */
+bool tvm_subplugin::configure_meta (GstTensorsInfo &tensor_meta, std::vector<tvm_data_info> &tensor_info_list)
+{
+  for (unsigned int i = 0; i < tensor_meta.num_tensors; ++i) {
+    tvm_data_info tvm_info;
+    auto &info = tensor_meta.info[i];
+    
+    tvm_info.shape = std::vector<int64_t> (std::begin(info.dimension), std::end(info.dimension));
+    if (!convert_nns_type (info.type, tvm_info.dtype)) {
+      nns_loge ("Failed to convert input type.");
+      return false;
+    }
+    tensor_info_list.push_back (tvm_info);
```

Comment:
is it okay that tensor_info_list is half filled?

otherwise how about create a temp vector and swap after it didn't return?

Reply:
I think it's ok if cleaned up afterwards, but the solution looks worthy. thx